src/infrastructure/weather/interfaces/controllers/weather.controller.ts

```
@@ -0,0 +1,20 @@
+import { Controller, Get, Inject, Param, UsePipes } from '@nestjs/common';
+import { WeatherCityValidationPipe } from 'src/common/pipes/city-validation.pipe';
+import {
+  WeatherInterface,
+  WeatherToken,
+} from 'src/infrastructure/weather/core/weather.interface';
+
+@Controller('internal/weather')
+export class WeatherInternalController {
+  constructor(
+    @Inject(WeatherToken)
+    private readonly weatherService: WeatherInterface,
+  ) {}
+
+  @UsePipes(WeatherCityValidationPipe)
```

Why do you have separate pipes for business rules? 
Shouldn't the logic from there be in services and throw business exceptions, such as 422 instead of 400?

---

I intentionally keep business validation in the pipe to fail fast and reject invalid cities before reaching the service.
In my findCity method, I throw a 404 error if the city is not found:

```ts
  async findCity(cityName: string): Promise<City> {
  ..//

    if (!response.results || response.results.length === 0) {
      throw new NotFoundException(`City "${cityName}" not found`);
    }

 ..//
  }